lib/menu.c

```diff
@@ -707,24 +749,128 @@ bm_menu_poll_key(struct bm_menu *menu, uint32_t *out_unicode)
     return key;
 }
 
+struct bm_pointer
+bm_menu_poll_pointer(struct bm_menu *menu)
+{
+    assert(menu);
+
+    struct bm_pointer pointer = {0};
+
+    if (menu->renderer->api.poll_pointer)
+        pointer = menu->renderer->api.poll_pointer(menu);
+
+    return pointer;
+}
+
+struct bm_touch
+bm_menu_poll_touch(struct bm_menu *menu)
+{
+    assert(menu);
+
+    struct bm_touch touch = {0};
+
+    if (menu->renderer->api.poll_touch)
+        touch = menu->renderer->api.poll_touch(menu);
+
+    return touch;
+}
+
+void
+bm_menu_release_touch(struct bm_menu *menu)
+{
+    assert(menu);
+
+    if (menu->renderer->api.release_touch)
+        menu->renderer->api.release_touch(menu);
+}
+
 static void
 menu_next(struct bm_menu *menu, uint32_t count, bool wrap)
 {
     if (menu->index < count - 1) {
-        menu->index++;
+        bm_menu_set_highlighted_index(menu, menu->index + 1);
     } else if (wrap) {
-        menu->index = 0;
+        bm_menu_set_highlighted_index(menu, 0);
     }
 }
 
 static void
 menu_prev(struct bm_menu *menu, uint32_t count, bool wrap)
 {
     if (menu->index > 0) {
-        menu->index--;
+        bm_menu_set_highlighted_index(menu, menu->index - 1);
     } else if (wrap) {
-        menu->index = count - 1;
+        bm_menu_set_highlighted_index(menu, count - 1);
+    }
+}
+
+static void
+menu_point_select(struct bm_menu *menu, uint32_t posx, uint32_t posy, uint32_t displayed)
+{
+    (void) posx;
+    uint32_t selected_line = posy / (bm_menu_get_height(menu) / displayed);
```

Comment:
this looks like division by zero can happen

Reply:
should be fixed !